cfg/Instructions.h

```diff
@@ -276,14 +273,16 @@ INSN(TAbsurd) : public Instruction {
     std::string toString(const core::GlobalState &gs, const CFG &cfg) const;
     std::string showRaw(const core::GlobalState &gs, const CFG &cfg, int tabs = 0) const;
 };
-CheckSize(TAbsurd, 32, 8);
+CheckSize(TAbsurd, 24, 8);
 
 class InstructionPtr final {
     using tagged_storage = uint64_t;
 
-    static constexpr tagged_storage TAG_MASK = 0xffff;
-
-    static constexpr tagged_storage PTR_MASK = ~TAG_MASK;
+    static constexpr tagged_storage TAG_MASK = 0xff;
```

Comment:
This seems a little bit like living on the edge? We already have 16 Instruction types (*and* it looks like we start the numbering at 1, so the 16th enum value’s integer wraps around and is 0). It doesn’t seem outlandish to me to think that we’ll need one more instruction kind at some point.

Reply:
This is a good callout, but the tag is not a bitmask: we can have ~200 tags and not overflow this field.

Reply:
Oh sorry, one f is 16, ff is 256 🤦‍♂️